### DISCUSSION TITLE
how to set value as empty if property or variable is not present

### DISCUSSION THREAD

**Discussion by takeAction** [jxlsteam/jxls#294] [Q&A] [answered]
jxls 2.x. I defined a property in excel  template like `${abc}`, this property may be not present in context. Now it prints out ${abc} in generated excel file if it is not in context. how could I set it as empty string if this property is not in context?

**Comment by SoltauFintel** [comment, marked as answer]
See [Clear template cells](https://jxls.sourceforge.net/builder.html) (3.0.0-beta1)

**Reply by takeAction** [reply]
Thanks @SoltauFintel , does any solutions in `jxls 2.x`? I am using 2.x and cannot upgrade to 3.0

**Reply by SoltauFintel** [reply]
I think this should also be possible in 2.x. But I don't know how to do it by heart. Please search the old documentation and look at the sources. Maybe TransformationConfig?

**Reply by takeAction** [reply]
Thanks for your tip. I checked `TransformationConfig`, there is `ExpressionEvaluator` take charge of evaluating expression, by default, jxls 2 will throw an exception if the expression cannot be evaluated and clear the cell. If we want to remove the exception we need a custom expression evaluator. In my case, there is wrong in `jx:area(lastCell="")` because my template has some merge cell.
